src/cascadia/TerminalApp/Pane.cpp

```diff
@@ -71,6 +71,127 @@ Pane::Pane(const GUID& profile, const TermControl& control, const bool lastFocus
     });
 }
 
+// Method Description:
+// - Extract the terminal settings from the current (leaf) pane's control
+//   to be used to create an equivalent control
+// Arguments:
+// - <none>
+// Return Value:
+// - Arguments appropriate for a SplitPane or NewTab action
+NewTerminalArgs Pane::GetTerminalArgsForPane() const
+{
+    // Leaves are the only things that have controls
+    assert(_IsLeaf());
+
+    NewTerminalArgs args{};
+    auto controlSettings = _control.Settings().as<TerminalSettings>();
+
+    args.Profile(controlSettings.ProfileName());
+    args.StartingDirectory(controlSettings.StartingDirectory());
+    args.TabTitle(controlSettings.StartingTitle());
+    args.Commandline(controlSettings.Commandline());
+    args.SuppressApplicationTitle(controlSettings.SuppressApplicationTitle());
+    if (controlSettings.TabColor() || controlSettings.StartingTabColor())
+    {
+
+        til::color c;
+        // StartingTabColor is prioritized over other colors
+        if (controlSettings.StartingTabColor())
+        {
+            c = til::color(controlSettings.StartingTabColor().Value());
+        }
+        else
+        {
+            c = til::color(controlSettings.TabColor().Value());
+        }
+
+        args.TabColor(winrt::Windows::Foundation::IReference<winrt::Windows::UI::Color>(c));

```

Comment:
I put this here, and it saves, but it also doesn't seem like creating a new tab actually uses the setting?

Reply:
Looks like the problem is actually just that when you set the tab color from the command palette doesn't set it on the TermControl?